File: app/src/main/java/pl/szymonchaber/donttouchit/screenblocking/view/BackBlockingFrameLayout.java
package pl.szymonchaber.donttouchit.screenblocking.view;

import android.content.Context;
import android.graphics.PixelFormat;
import android.support.annotation.AttrRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.KeyEvent;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.FrameLayout;

public class BackBlockingFrameLayout extends FrameLayout {

    public BackBlockingFrameLayout(@NonNull Context context) {
        super(context);
    }

    public BackBlockingFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public BackBlockingFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs, @AttrRes int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean dispatchKeyEvent(KeyEvent event) {
        return event != null && event.getKeyCode() == KeyEvent.KEYCODE_BACK || super.dispatchKeyEvent(event);
    }

    public ViewGroup.LayoutParams createParams() {
        WindowManager.LayoutParams params =
                new WindowManager.LayoutParams(WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT,
                        WindowManager.LayoutParams.TYPE_PHONE, WindowManager.LayoutParams.FLAG_FULLSCREEN,
                        PixelFormat.TRANSLUCENT);
        params.gravity = Gravity.CENTER;
        return params;
    }
}

File: app/src/main/kotlin/pl/szymonchaber/donttouchit/screenblocking/view/BackBlockingFrameLayout.kt
package pl.szymonchaber.donttouchit.screenblocking.view

import android.content.Context
import android.graphics.PixelFormat
import android.support.annotation.AttrRes
import android.util.AttributeSet
import android.view.Gravity
import android.view.KeyEvent
import android.view.ViewGroup
import android.view.WindowManager
import android.widget.FrameLayout

class BackBlockingFrameLayout : FrameLayout {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, @AttrRes defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun dispatchKeyEvent(event: KeyEvent?): Boolean {
        return when (event?.keyCode) {
            KeyEvent.KEYCODE_BACK -> true
            else                  -> super.dispatchKeyEvent(event)
        }
    }

    fun createParams(): ViewGroup.LayoutParams {
        return WindowManager.LayoutParams(
            WindowManager.LayoutParams.MATCH_PARENT, WindowManager.LayoutParams.MATCH_PARENT,
            WindowManager.LayoutParams.TYPE_PHONE, WindowManager.LayoutParams.FLAG_FULLSCREEN,
            PixelFormat.TRANSLUCENT
        ).apply {
            gravity = Gravity.CENTER
        }
    }
}
